### PR TITLE
[SPARK-125] Handle and test for two edge cases for confused clients

### PR DESCRIPTION
**Pull request by Padawan00123**

### CODE REVIEW

File: tests/unit/test_ratmama.py [hunk 1]
```diff
@@ -178,6 +206,24 @@ async def test_manual_signal(bot_fx,
...
+        async_callable_fx,
+        monkeypatch):
+
+    context = await Context.from_message(
+        bot_fx,
+        "#unit_test",
+        "unit_test",
+        "Ratsignal some_service")
+
+    monkeypatch.setattr(context, "reply", async_callable_fx)
```
Comment (theunkn0wn1) [inline tests/unit/test_ratmama.py]: [question] why was the reply attribute patched here? I don't see any usage of `async_callable_fx` used in an assertion as a result of this patch?

File: src/packages/ratmama/ratmama_parser.py [hunk 2]
```diff
@@ -108,6 +109,16 @@ async def handle_ratmama_announcement(ctx: Context) -> None:
     lang_code: str = result.group("language_code")
     nickname: Optional[str] = result.group("nick")
 
+    client = await User.from_pydle(ctx.bot, client_name)
+
+    if client is not None:
+        client_hostname = client.hostname
```
Comment (theunkn0wn1) [inline src/packages/ratmama/ratmama_parser.py]: [Error] Redundant variable assignment

You needlessly assign an instance attribute of a dataclass to a local name (read only, cheap retrieval) , it is better NOT to extract that one attribute from the object, as it may lead to broken expectations.

File: src/packages/ratmama/ratmama_parser.py [hunk 3]
```diff
@@ -108,6 +109,16 @@ async def handle_ratmama_announcement(ctx: Context) -> None:
...
+
+        if client_hostname in (
+                "services.fuelrats.com",
+                "bot.fuelrats.com"):
+            return
```
Comment (theunkn0wn1) [inline src/packages/ratmama/ratmama_parser.py]:
```suggestion
    if client and client.hostname in ("services.fuelrats.com", "bot.fuelrats.com"):
        logger.warning("attempted to create rescue for a service!")
        # emit a warning to the dispatch, so they can handle this themselves.
        return await ctx.reply("signal attempted to create rescue for a service. Dispatch: please manually inject!")
```

**Review by theunkn0wn1 — CHANGES_REQUESTED** [review]
Looks ok, but we need to warn the dispatch when we fail in this mode so they can handle it themselve.

**Review by theunkn0wn1 — DISMISSED** [review]
LGTM :+1:

**Review by theunkn0wn1 — APPROVED** [review]
LGTM :+1:
